app.rb

```diff
@@ -0,0 +1,177 @@
+      end
+    end
+  end
+
+  def list_all_people
+    if @people.empty?
+      puts 'No person added to the system'
+    else
+      @people.each do |person|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def list_all_people_with_index
+    @people.each_with_index do |person, index|
+      puts "#{index}. [#{person.class}] Name: #{person.name}, ID #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp.downcase
+
+    case permission
+    when 'y'
+      parent_permission = true
+    when 'n'
+      parent_permission = false
+    end
+
+    @people.push(Student.new('11', age, name, parent_permission: parent_permission))
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    @people.push(Teacher.new(specialization, age, name))
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+
+    case person_type
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    end
+
+    puts 'Person created successfully'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    @books.push(Book.new(title, author))
+
+    puts 'Book Created Successfully'
+  end
+
+  def create_rental()
+    puts 'Select a book from the following list by number'
+    list_all_books
+
+    book_index = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not id)'
+    list_all_people_with_index
+
+    person_index = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp
+
+    Rental.new(date, @books[book_index], @people[person_index])
+
+    puts 'Rental created successfully'
+  end
+
+  def list_all_rental
+    puts 'All people in the system'
+    list_all_people
+    print 'Id of the person: '
+    id = gets.chomp.to_i
+    person_array = @people.select { |person| person.id == id }
+
+    if person_array.empty?
+      puts 'No person matches the given ID!'
+    else
+      person = person_array[0]
+      if person.rentals.empty?
+        puts 'This person does not have any rentals'
+      else
+        person_array[0].rentals.each do |rental|
+          puts "Date #{rental.date}, Book: #{rental.book.title} Author: #{rental.book.author}"
+        end
+      end
+    end
+  end
+
+  def choose_option(input)
+    case input
+    when '1' then list_all_books
+    when '2' then list_all_people
+    when '3' then create_person
+    when '4' then create_book
+    when '5' then create_rental
+    when '6' then rentals_of_person
+    end
+  end
```

Comment:
- [ ] Great job applying the requested changes! 💯 However, there's one more (tiny) error preventing a successful app execution:

I'm getting this error when trying to list all rentals.

![carbon (3)](https://user-images.githubusercontent.com/77069011/218201297-9099055e-e270-4714-962c-9310eacb94a5.png)

However, it is mostly noise, the error is a typo in your code: 🤖 

```suggestion
  def choose_option(input)
    case input
    when '1' then list_all_books
    when '2' then list_all_people
    when '3' then create_person
    when '4' then create_book
    when '5' then create_rental
    when '6' then list_all_rental
    end
  end
```

Please consider applying this change! 🤓 🔧